src/main/java/org/javamodularity/moduleplugin/tasks/MergeClassesTask.java

```diff
@@ -0,0 +1,39 @@
+package org.javamodularity.moduleplugin.tasks;
+
+import org.gradle.api.Project;
+import org.gradle.api.Task;
+import org.gradle.api.plugins.ApplicationPlugin;
+import org.gradle.api.plugins.JavaPlugin;
+import org.gradle.api.specs.Spec;
+import org.javamodularity.moduleplugin.extensions.CompileModuleOptions;
+
+import java.util.List;
+
+public class MergeClassesTask extends AbstractModulePluginTask {
+    public MergeClassesTask(Project project) {
+        super(project);
+    }
+
+    public void configureMergeClasses() {
+        var mergeClasses = helper().mergeClassesTask();
+
+        mergeClasses.into(helper().getMergedDir());
+
+        mergeClasses.onlyIf(new Spec<Task>() {
```

Comment:
I wonder - can this be converted to a lambda? As far as I remember, only `doFirst` and `doLast` are cached as task inputs and are affected by this "cannot be a lambda" problem. But again, I'm not 100% sure.

Reply:
Yes, it seems it can be converted.